test/language/expressions/async-generators/early-errors-next-receiver-not-async-iterator.js

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Benjamin Gruenbaum <benjamingr@gmail.com>
+esid: pending
+description: >
+  next() rejects with a TypeError when receiver is not an AsyncGenerator instance:
+  phase: early
+  type: TypeError
+---*/
+
+(async function*() {})().next.call((function*() {})()).then(assert.throw.bind(assert), function(err) {
```

Comment:
There's probably a better way to assert that the resolve handler is never called. Maybe just throw a Teat262Assertion in the handler indicating that it shouldn't be called, it's easier to understand